Underanalyzer/Decompiler/GlobalFunctions.cs

```diff
@@ -121,70 +133,71 @@ private string GetFunctionNameAfterFragment(Block block, out IGMFunction foundFu
+                        ..
+                    ])
+                {
+                    // We have a name!
+                    return funcName;
                 }
+                break;
+            }
             case Opcode.Call:
+            {
+                // This is a struct or constructor function
+                if (block.Instructions is not
+                    [
+                        _, _,
+                    { Kind: Opcode.Call, Function.Name.Content: VMConstants.NullObjectFunction },
+                    { Kind: Opcode.Call, Function.Name.Content: VMConstants.MethodFunction },
+                        ..
+                    ])
                 {
-                    // This is a struct or constructor function
-                    if (block.Instructions is not
-                        [
-                            _, _,
-                        { Kind: Opcode.Call, Function.Name.Content: VMConstants.NullObjectFunction },
-                        { Kind: Opcode.Call, Function.Name.Content: VMConstants.MethodFunction },
-                            ..
-                        ])
-                    {
-                        // Failed to match instructions
-                        return null;
-                    }
+                    // Failed to match instructions
+                    return null;
+                }
 
-                    // Check if we're a struct or function constructor (named)
-                    if (block.Instructions is
-                        [
-                            _, _, _, _,
-                        { Kind: Opcode.Duplicate, DuplicationSize2: 0 },
-                        { Kind: Opcode.PushImmediate, ValueShort: short pushVal },
-                        { Kind: Opcode.Pop, Variable.Name.Content: string funcName },
-                            ..
-                        ])
+                // Check if we're a struct or function constructor (named)
+                if (block.Instructions is
+                    [
+                        _, _, _, _,
+                    { Kind: Opcode.Duplicate, DuplicationSize2: 0 },
+                    { Kind: Opcode.PushImmediate, ValueShort: short pushVal },
+                    { Kind: Opcode.Pop, Variable.Name.Content: string funcName },
+                        ..
+                    ])
+                {
+                    // Check if struct or constructor
+                    if (pushVal != -16 && pushVal != -5)
                     {
-                        // Check if struct or constructor
-                        if (pushVal != -16 && pushVal != -5)
-                        {
-                            // We're a constructor!
-                            return funcName;
-                        }
+                        // We're a constructor!
+                        return funcName;
                     }
-                    break;
                 }
+                break;
+            }
+            // TODO: default case?
```

Comment:
This is just instruction pattern matching, it falls through to the `return null;` afterwards if none of the conditions fully succeed.

Reply:
not what i meant. the todo was specifically referring to "Explicitly document all **expected** paths, and reserve default for unexpected value"